### PR TITLE
prayer: remove

### PR DESCRIPTION
**Pull request by tu-maurice**
## Description of changes

This PR removes anything related to the prayer package. For this I present the following reasons:

- The package has not been touched in almost 4 years.
- It failed to build in the past 2 years, which means no active users.
- The sources are not available anymore, so we're only relying on hydras cache for this.
- Fixing the build doesn't get the derivation into a clean nor usable state. See #266129 

I think this package is beyond a point where marking it as broken is reasonable, because "Such unconditionally-broken packages should be removed from Nixpkgs eventually unless they are fixed." (quote Nixpkgs manual) and the package is beyond fixing.

I should probably add some error message somewhere, explaining that the packages is gone, but I have no idea how or where. Any thoughts?

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ambroisie** [comment]
The place to add a message about removed packages is in `aliases.nix`, you should find plenty of examples in the file.

**Comment by tu-maurice** [comment]
Neat. Is there a similar spot for the module? Or do I just remove it.

**Comment by ambroisie** [comment]
NixOS modules have `rename.nix` I believe (can't check as I'm on mobile).

**Comment by tu-maurice** [comment]
Ah. Yeah there's one with plenty examples. Thank you, @ambroisie :)

**Review by ambroisie — APPROVED** [review]

**Comment by pbsds** [comment]
I think removal of service modules should go in the release notes.

**Review by ambroisie — APPROVED** [review]

File: nixos/modules/misc/ids.nix [hunk 1]
```diff
@@ -86,7 +86,6 @@ in
       #rtkit = 45; # dynamically allocated 2021-09-03
       dovecot2 = 46;
       dovenull2 = 47;
-      prayer = 49;
       mpd = 50;
```
Comment (pbsds) [inline nixos/modules/misc/ids.nix:89]:
```suggestion
      # prayer = 49 # dropped in 23.11
      mpd = 50;
```

File: nixos/modules/misc/ids.nix [hunk 2]
```diff
@@ -411,7 +410,6 @@ in
       #rtkit = 45; # unused
       dovecot2 = 46;
       dovenull2 = 47;
-      prayer = 49;
       mpd = 50;
```
Comment (pbsds) [inline nixos/modules/misc/ids.nix:413]:
```suggestion
      # prayer = 49 # dropped in 23.11
      mpd = 50;
```

**Review by pbsds — APPROVED** [review]
just a small nit

File: nixos/modules/misc/ids.nix (hunk 1 above)
Comment (tu-maurice) [reply nixos/modules/misc/ids.nix:90]: Ack.

File: nixos/modules/misc/ids.nix (hunk 2 above)
Comment (tu-maurice) [reply nixos/modules/misc/ids.nix:415]: Ack

**Comment by AndersonTorres** [comment]
A prayer for the removal of prayer.

**Pull request merged**